backend/routes/users.js

```
@@ -402,4 +402,124 @@ router.get('/ratings/:userId/:partnerId', async (req, res) => {
   }
 });
 
+router.post('/match-requests', async (req, res) => {
+  try {
+    const { senderId, receiverId } = req.body;
+
+    if (!senderId || !receiverId) {
+      return res.status(400).json({ error: 'Both sender and receiver IDs are required' });
+    }
+
+    if (senderId === receiverId) {
+      return res.status(400).json({ error: 'Cannot send match request to yourself' });
+    }
+
+    const existingMatch = await prisma.match.findFirst({
+      where: {
+        OR: [
+          { user1Id: parseInt(senderId), user2Id: parseInt(receiverId) },
+          { user1Id: parseInt(receiverId), user2Id: parseInt(senderId) }
+        ]
+      }
+    });
+
+    if (existingMatch) {
+      return res.status(409).json({ error: 'Match already exists' });
+    }
+
+    const existingRequest = await prisma.matchRequest.findFirst({
+      where: {
+        OR: [
+          { senderId: parseInt(senderId), receiverId: parseInt(receiverId) },
+          { senderId: parseInt(receiverId), receiverId: parseInt(senderId) }
+        ]
+      }
+    });
+
+    if (existingRequest) {
+      return res.status(409).json({ error: 'Match request already exists' });
+    }
```

Hmm I am not sure about this. Let's make up an example between two users called Alice and Bob

First Alice sees Bob in her app, decides to swipe right on him, so we create a matchRequest with Alice is the sender and Bob is the receiver. If Alice somehow sees Bob again and swipes right, then yes an error should be thrown because match request already exist

But now lets say Bob is using the app, and before he goes to the Matches tab he sees Alice recommended to him and decides to swipe right. So now we make a new matchRequest with Bob is the sender and Alice is the receiver. In this case, I don't think we should throw a failure saying match request already exist. Instead we should perform the logic to make this an actual match between Alice and Bob.

What do you think, is there something I am missing?